mypy/test/testcheck.py

```diff
@@ -170,15 +170,22 @@ def run_case_once(self, testcase: DataDrivenTestCase, incremental_step: int = 0)
             # Always set to none so we're forced to reread the module in incremental mode
             sources.append(BuildSource(program_path, module_name,
                                        None if incremental_step else program_text))
+        streamed_messages = []
+
+        def flush_errors(msgs: List[str], serious: bool) -> None:
+            streamed_messages.extend(msgs)
+
         res = None
         try:
             res = build.build(sources=sources,
                               options=options,
-                              alt_lib_path=test_temp_dir)
+                              alt_lib_path=test_temp_dir,
+                              flush_errors=flush_errors)
             a = res.errors
         except CompileError as e:
             a = e.messages
         a = normalize_error_messages(a)
+        streamed_messages = normalize_error_messages(streamed_messages)
```

Comment:
I was quite surprised that this could work. Until I realized that `BuildResult` calls `errors.messages()` which returns *all* messages (redundantly). This complexity increase worries me again).